programs/stake_api/src/stake_instruction.rs

```diff
@@ -164,28 +164,20 @@ pub fn create_stake_account_and_delegate_stake(
     instructions
 }
 
-// for instructions that whose authorized signer may differ from the account's pubkey
-fn metas_for_authorized_signer(
-    account_pubkey: &Pubkey,
-    authorized_signer: &Pubkey, // currently authorized
-    other_params: &[AccountMeta],
+// for instructions whose authorized signer may already be in account parameters
+fn metas_with_signer(
+    mut metas: Vec<AccountMeta>, // parameter metas, in order
+    signer: &Pubkey,             // might already appear in paramters
 ) -> Vec<AccountMeta> {
-    let is_own_signer = authorized_signer == account_pubkey;
-
-    // vote account
-    let mut account_metas = vec![AccountMeta::new(*account_pubkey, is_own_signer)];
-
-    for meta in other_params {
-        account_metas.push(meta.clone());
-    }
-
-    // append signer at the end
-    if !is_own_signer {
-        account_metas.push(AccountMeta::new_credit_only(*authorized_signer, true))
-        // signer
+    for meta in metas.iter_mut() {
```

Comment:
Weird that clippy didn't complain here. I'd expect it to say to use `&mut metas` instead of `metas.iter_mut()`.